andela/ah-leagueOfLegends
# #159965488 Send notifications to users.

## What does this PR do?
```
Enables subscribed users to receive email, in-app notifications after a new article is published by the author they follow or any comments are made in the article that the user has favourited.
```

## Description of Task to be completed?
```
-  A user will be able to receive notifications when users he/she follows publish new articles.
-  A user will able to receive notifications when articles he/she favourited have new interactions 
      (like     new comments).
- Also, a user will be able to opt in and out of these notifications.

```

## How should this be manually tested?

```
- Clone this repo.
- Create a Virtual environment and pip install the requirements.txt
- Register and login at least three users.
- One user should create an article and the other follows the author.
- Any of the logged in users should comment to the article.
- Users will receive email notifications, both in-app and via email.

```

## What are the relevant pivotal tracker stories?

[#159965488](https://www.pivotaltracker.com/story/show/159965488)


## Screenshots if available to support PR.

### Getting In-app notifications
<img width="1440" alt="screen shot 2018-09-20 at 20 50 50" src="https://user-images.githubusercontent.com/39088834/45836991-f01a8480-bd16-11e8-8f44-0ea5f6230d07.png">

### Getting Email notifications
#### New Article
<img width="830" alt="screen shot 2018-09-20 at 20 52 14" src="https://user-images.githubusercontent.com/39088834/45837085-2a842180-bd17-11e8-873f-d57b9d4e1b3b.png">

#### New Comment

<img width="830" alt="screen shot 2018-09-20 at 20 53 50" src="https://user-images.githubusercontent.com/39088834/45837156-569fa280-bd17-11e8-837e-8d7590a99ba8.png">



## Questions:

```
Was this PR helpful?

```
